spec/index.bs

```diff
@@ -4625,19 +4669,58 @@ interface mixin GPUProgrammablePassEncoder {
 
             **Arguments:**
             <pre class=argumentdef for="GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsetsData, dynamicOffsetsDataStart, dynamicOffsetsDataLength)">
-                index:
-                bindGroup:
-                dynamicOffsetsData:
-                dynamicOffsetsDataStart:
-                dynamicOffsetsDataLength:
+                |index|: The index to set the bind group at.
+                |bindGroup|: Bind group to use for subsequent render or compute commands.
+                |dynamicOffsetsData|: Array containing buffer offsets in bytes for each entry in
+                    |bindGroup| with marked as {{GPUBindGroupLayoutEntry/hasDynamicOffset}}.
+                |dynamicOffsetsDataStart|: Offset in elements into |dynamicOffsetsData| where the
+                    buffer offset data begins.
+                |dynamicOffsetsDataLength|: Number of buffer offsets to read from |dynamicOffsetsData|.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsetsData, dynamicOffsetsDataStart, dynamicOffsetsDataLength)}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|.{{GPUObjectBase/[[device]]}}:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - |bindGroup| is [$valid to use with$] |this|.
+                        - |index| &lt; {{GPULimits/maxBindGroups|GPULimits.maxBindGroups}}.
+                        - |dynamicOffsetsDataLength| is
+                            |bindGroup|.{{GPUBindGroup/[[layout]]}}.{{GPUBindGroupLayout/[[dynamicOffsetCount]]}}.
+                        - |dynamicOffsetsDataStart| + |dynamicOffsetsDataLength| &le; |dynamicOffsetsData|.length.
+
+                        - [$Iterate over each dynamic binding offset$] in |bindGroup| and
+                            run the following steps for each |bufferBinding|, |minBufferBindingSize|,
+                            and |dynamicOffsetIndex|:
+
+                            - Let |bufferDynamicOffset| be
+                                |dynamicOffsetsData|[|dynamicOffsetIndex| + |dynamicOffsetsDataStart|].
+                            - |bufferBinding|.{{GPUBufferBinding/offset}} + |bufferDynamicOffset| +
+                                |minBufferBindingSize| &le;
+                                |bufferBinding|.{{GPUBufferBinding/buffer}}.{{GPUBuffer/[[size]]}}.
```

Comment:
In the definition of the `GPUBindGroupLayoutEntry` it says that `minBufferBindingSize` defaults to `0` for buffer types. If that's the case then this works out regardless of if it's provided or not, right?

Reply:
Hmmm, I think this was unintentional. It looks like it was effectively a merge conflict between #874 and #919. minBufferBindingSize should not have a default (`undefined` is a special/sentinel value that says the size will be checked at draw time instead of createBindGroup time).